clang/include/clang/Lex/LiteralSupport.h

```diff
@@ -246,18 +249,19 @@ class StringLiteralParser {
   StringLiteralEvalMethod EvalMethod;
 
 public:
-  StringLiteralParser(ArrayRef<Token> StringToks, Preprocessor &PP,
-                      StringLiteralEvalMethod StringMethod =
-                          StringLiteralEvalMethod::Evaluated);
+  StringLiteralParser(
+      ArrayRef<Token> StringToks, Preprocessor &PP,
+      StringLiteralEvalMethod StringMethod = StringLiteralEvalMethod::Evaluated,
+      ConversionAction Action = ToExecCharset);
```

Comment:
Why do we need Conversion at all.
I would expect that any ordinary, non-unevaluated literal would be encoded and the
`LiteralConverter` should be the same for all strings so it can live in `Preprocessor`

Reply:
So I put up a table on my old RFC https://discourse.llvm.org/t/rfc-enabling-fexec-charset-support-to-llvm-and-clang-reposting/71512 which shows the different encoding needed depending on the context. So not all strings will be translated to the execution encoding. A separate PR after this one will be needed to ensure we use the correct encoding depending on the context.